app/questionnaire/questionnaire_schema.py

```diff
@@ -14,75 +14,88 @@ def __init__(self, questionnaire_json, language_code=DEFAULT_LANGUAGE_CODE):
         self.language_code = language_code
         self._parse_schema()
 
-    @property
-    def sections(self):
+    def get_sections(self):
         return self._sections_by_id.values()
 
-    @property
-    def groups(self):
+    def get_section_via_section_id(self, section_id: str):
+        return self._sections_by_id.get(section_id)
+
+    def get_section_via_group_id(self, group_id: str):
+        group = self.group(group_id)
+
+        try:
+            return self.get_section_via_section_id(group['parent_id'])
+        except KeyError:
+            return None
+
+    def get_section_via_block_id(self, block_id: str):
+        block = self.block(block_id)
+
+        try:
+            group_id = block['parent_id']
+        except TypeError:
+            return None
+
+        try:
+            section_id = self.group(group_id)['parent_id']
+        except TypeError:
```

Comment:
I'm slightly confused by this, when is the group == `None`?

Reply:
When an incorrect block id is used.